rtgui/imagearea.cc

```diff
@@ -170,10 +169,11 @@ CropWindow* ImageArea::getCropWindow (int x, int y)
 
     CropWindow* cw = mainCropWindow;
 
-    for (std::list<CropWindow*>::iterator i = cropWins.begin(); i != cropWins.end(); i++)
-        if ((*i)->isInside (x, y)) {
-            return *i;
+    for (auto cropWin : cropWins) {
```

Comment:
Yeah, that is the reason that for example Java explicitly decided against something like `auto`. AFAIK, the idea of the C++ standard committee was that the concrete type shouldn't matter to the programmer. It should even change automatically if the right hand side of the assignment changes. If there is a problem, it should be caught later on due to type constraints.
At least with `auto` the will never be narrowing, loss of precision or unintended wasteful implicit casts. But for example, for me personally using `auto` somehow felt wrong in the beginning and only after using it for a while I began to feel comfortable about it.